cypress-tests/cypress/support/redirectionHandler.js

```diff
@@ -0,0 +1,319 @@
+          cy.wait(5000);
+          cy.url().should("include", "status=succeeded");
+          cy.wait(5000);
+          break;
+        case "sofort":
+          cy.get(".modal-overlay.modal-shown.in", { timeout: 5000 }).then(
+            ($modal) => {
+              // If modal is found, handle it
+              if ($modal.length > 0) {
+                cy.get("button.cookie-modal-deny-all.button-tertiary")
+                  .should("be.visible")
+                  .should("contain", "Reject All")
+                  .click({ force: true, multiple: true });
+                cy.get("div#TopBanks.top-banks-multistep")
+                  .should("contain", "Demo Bank")
+                  .as("btn")
+                  .click();
+                cy.get("@btn").click();
+              } else {
+                cy.get("input.phone").type("9123456789");
+                cy.get("#button.onContinue")
+                  .should("contain", "Continue")
+                  .click();
+              }
+            }
+          );
+          break;
+        case "trustly":
+          break;
+        default:
+          throw new Error(
+            `Unsupported payment method type: ${payment_method_type}`
+          );
+      }
+      verifyUrl = true;
+      break;
+    case "trustpay":
+      switch (payment_method_type) {
+        case "eps":
+          cy.get("._transactionId__header__iXVd_").should(
+            "contain.text",
+            "Bank suchen ‑ mit eps zahlen."
+          );
+          cy.get(".BankSearch_searchInput__uX_9l").type(
+            "Allgemeine Sparkasse Oberösterreich Bank AG{enter}"
+          );
+          cy.get(".BankSearch_searchResultItem__lbcKm").click();
+          cy.get("._transactionId__primaryButton__nCa0r").click();
+          cy.get("#loginTitle").should(
+            "contain.text",
+            "eps Online-Überweisung Login"
+          );
+          cy.get("#user")
+            .should("be.visible")
+            .should("be.enabled")
+            .focus()
+            .type("Verfügernummer");
+          cy.get("input#submitButton.btn.btn-primary").click();
+          break;
+        case "ideal":
+          cy.get("p").should(
+            "contain.text",
+            "Choose your iDeal Issuer Bank please"
+          );
+          cy.get("#issuerSearchInput").click();
+          cy.get("#issuerSearchInput").type("ING{enter}");
+          cy.get("#trustpay__selectIssuer_submit").click();
+          break;
+        case "giropay":
+          cy.get("._transactionId__header__iXVd_").should(
+            "contain.text",
+            "Bank suchen ‑ mit giropay zahlen."
+          );
+          cy.get(".BankSearch_searchInput__uX_9l").type(
+            "Volksbank Hildesheim{enter}"
+          );
+          cy.get(".BankSearch_searchIcon__EcVO7").click();
+          cy.get(".BankSearch_bankWrapper__R5fUK").click();
+          cy.get("._transactionId__primaryButton__nCa0r").click();
```

Comment:
Won't these identifiers change over time?

Reply:
I believe not. Other simpler methods never worked here when I tried. Had to resort to the work that @tsdk02 and @Abhitator216 had done.